R/zchunk_L261.Cstorage.R

```diff
@@ -56,204 +70,308 @@ module_energy_L261.Cstorage_DISABLED <- function(command, ...) {
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # Silence package notes
+    . <- available <- capacity.factor <- curr_table <-
+      extractioncost <- grade <- logit.type <- minicam.energy.input <-
+      minicam.non.energy.input <- `output-unit` <- `price-unit` <- resource <-
+      resource_type <- share.weight <- subresource <- subsector <- subsector.name <-
+      supplysector <- technology <- value <- year <- region <- depresource <-
+      output.unit <- price.unit <- market <- NULL
+
+    # A
+    # Create tables for carbon storage resource information
+    # A61.rsrc_info provides carbon storage resource info (output unit, price unit, capacity factor, market, etc)
+    A61.rsrc_info %>%
+      # Expand table to incorporate GCAM region names (use ID to ensure correct region ordering)
+      # We will use these specific region names to replace the broad term, regional, in the market column.
+      repeat_add_columns(GCAM_region_names) %>%
+      # Reset regional markets to the names of the specific regions
+      mutate(market = replace(market, market == "regional", region[market == "regional"])) %>%
+      rename(output.unit = `output-unit`, price.unit = `price-unit`) ->
+      L261.rsrc_info
+
+    # Split different types of resources into separate tables
+
+    # Create table reporting carbon storage information for unlimited resources only
+    L261.rsrc_info %>%
+      filter(resource_type == "unlimited-resource") %>%
+      select(region, unlimited.resource = resource, output.unit, price.unit, market, capacity.factor) ->
+      L261.UnlimitRsrc # This is a final ouput table.
+
+    # Create table reporting carbon storage information for depletable resources only
+    L261.rsrc_info %>%
+      filter(resource_type == "depresource") %>%
+      select(region, depresource = resource, output.unit, price.unit, market) ->
+      L261.DepRsrc # This is a final ouput table.
+
+
+    # B
+    # Supply curves of carbon storage resources
+    # L161.RsrcCurves_MtC_R reports carbon storage resource supply curves by GCAM region.
+    L161.RsrcCurves_MtC_R %>%
+      # Match in GCAM region names using region ID
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      select(region, depresource = resource, subresource, grade, available, extractioncost) ->
+      L261.DepRsrcCurves_C # This is a final output table.
+
+    # Calculate three different supply curves of carbon storage resources: high, low, and lowest.
+    # Multiply the extraction cost by their respective cost multipliers.
```

Comment:
I have no idea how these values were chosen, but they were created for the SSP scenarios as described in energy-data/assumptions/A_rscr_data.R.  I'm not sure whether the originator would have a justification for these values.
Please add an additional comment that they were created for the SSPs and that high, low and lowest is the level of CCS use and not cost. The multipliers represent increases to the CCS cost, for high, low and lowest levels of CCS use.